arch/arm/mach-imx/fiohab.c

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2019 Foundries.io
+ */
+
+#include <common.h>
+#include <config.h>
+#include <fuse.h>
+#include <mapmem.h>
+#include <image.h>
+#include <asm/io.h>
+#include <asm/system.h>
+#include <asm/arch/clock.h>
+#include <asm/arch/sys_proto.h>
+#include <asm/mach-imx/hab.h>
+
+#ifdef CONFIG_MX7ULP
+#define SRK_FUSE_BANK		(5)
+#define SRK_FIRST_FUSE		(0)
+#define SRK_NBR_FUSE		(8)
+#define SECURE_FUSE_BANK	(29)
+#define SECURE_FUSE_WORD	(6)
+#define SECURE_FUSE_VALUE	(0x80000000)
+#else
+#error "SoC not supported"
+#endif
+
+static hab_rvt_report_status_t *hab_check;
+
+static int hab_status(void)
+{
+	hab_check = (hab_rvt_report_status_t *) HAB_RVT_REPORT_STATUS;
+	enum hab_config config = 0;
+	enum hab_state state = 0;
+
+	if (hab_check(&config, &state) != HAB_SUCCESS) {
+		puts("HAB events active\n");
+		return 1;
+	}
+
+	return 0;
+}
+
+static int do_fiohab_close(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
+{
+	uint32_t fuse, fuse_env;
+	char fuse_name[20];
+	int i, j;
+	int ret;
+
+	if ((argc != 1)) {
+		cmd_usage(cmdtp);
+		return 1;
+	}
+
+	if (imx_hab_is_enabled()) {
+		printf("secure boot already enabled\n");
+		return 0;
+	}
+
+	if (hab_status())
+		return 1;
```

Comment:
Probably good to add another printf message here saying that the device got a HAB event and can't move forward.

At the same time I'm wondering if this check helps when the device is still missing the SRK hashes, as that won't mean much. The main issue is when you fuse SRK, reboot and get a hab event, as that indicates we shouldn't close the device.

Reply:
I dont think we need that printf there since is taken care of by the function itself (indicating active events).